src/main/kotlin/net/maizegenetics/phgv2/cli/CreateFastaFromHvcf.kt

```diff
@@ -66,33 +69,55 @@ class CreateFastaFromHvcf : CliktCommand( help = "Create a FASTA file from a h.v
      */
-    fun buildFastaFromHVCF(dbPath: String, outputFile: String, fastaType:String, hvcfDir: String ,hvcfFile : String, condaEnvPrefix:String) {
-        if(hvcfDir != "") {
-            //Loop through the directory and figure out which files are hvcf files
-            // the gvcf and hvcf files may be in the same folder, so verify specific extension
-            val hvcfFiles = File(hvcfDir).listFiles { file -> file.extension == "h.vcf" || file.name.endsWith("h.vcf.gz") || file.name.endsWith("hvcf") || file.name.endsWith("hvcf.gz")}
-            //Loop through each file and run the buildFastaFromHVCF function
-            BufferedWriter(FileWriter(outputFile)).use { output ->
-                writeSequences(output,hvcfFiles.flatMap { processSingleHVCF(VCFFileReader(it,false), dbPath,condaEnvPrefix) }
-                    .associate { it.id to it }
-                    .values.toList(), fastaType)
+    fun buildFastaFromHVCF(dbPath: String, outputDir: String, fastaType:String, hvcfDir: String ,hvcfFile : String, condaEnvPrefix:String) {
+
+        if (hvcfDir != "") {
+            // Loop through the directory and figure out which files are hvcf files
+            // The gvcf and hvcf files may be in the same folder, so verify specific extension
+            val hvcfFiles = File(hvcfDir).listFiles { file ->
+                file.name.endsWith("h.vcf") || file.name.endsWith("h.vcf.gz") || file.name.endsWith("hvcf") ||
+                        file.name.endsWith("hvcf.gz")
+            }
+
+            // Loop through each file and run the processSingleHVCF function
+            hvcfFiles?.forEach { hvcfFile ->
+                val tempFileName = hvcfFile.name.replace(HVCF_PATTERN, ".fa")
+                val outputFileName = File(tempFileName).name.replace(".fa", "_${fastaType}.fa")
+                val outputFile = "$outputDir/$outputFileName"
+
+                BufferedWriter(FileWriter(outputFile)).use { output ->
+                    writeSequences(output,
+                        processSingleHVCF(VCFFileReader(hvcfFile, false), dbPath, condaEnvPrefix)
+                            .associate { it.id to it }
+                            .values.toList(),
+                        fastaType)
+                }
             }
         }
-        else  if(hvcfFile == "") {
-            //Load in the TileDB
+        else if (hvcfFile == "") {
+            // Load in the TileDB
             TODO("TileDB VCF Reader Not implemented yet.  Please run with --hvcf-file")
         } else {
-            //Load in the HVCF
-            val hvcfFile = VCFFileReader(File(hvcfFile),false)
+            // Load in the HVCF
+            val hvcfFileReader = VCFFileReader(File(hvcfFile), false)
+            val tempFileName = File(hvcfFile).name.replace(HVCF_PATTERN, ".fa")
+            val outputFileName = File(tempFileName).name.replace(".fa", "_${fastaType}.fa")
+            //val outputFileName = "${fileHvcf.nameWithoutExtension}.fa"
```

Comment:
Remove comment.  Also similar to above can we get rid of tempFileName

Reply:
updated